### ISSUE TITLE
Update Android parse module to use FCM instead of GCM

### ISSUE THREAD

**Issue by antoinevg** [trigger-corp/forge#6]
As of April 10, 2018, Google has deprecated GCM. The GCM server and client APIs are deprecated and will be removed as soon as April 11, 2019. 

**Describe the solution you'd like**

### Platform Changes

- [x] Add support for Google Services gradle plugin
- [x] Add support for conditionals in `build_steps.json`
- [x] Add support for `google-services.json`
- [x] Add flag for clear text traffic on Android: `android_uses_cleartext_traffic`

### Module Changes

- [x] Delete config fields that are no longer relevant
- [x] Update parse module to `1.18.5`
- [x] Port ForgePushBroadcastReceiver to FCM
- [x] Fix custom icons field
- [x] Documentation for migration

**Comment by scthi** [comment]
@antoinevg We already did that, see: https://github.com/trigger-corp/trigger.io-parse/compare/master...scthi:RE-26-switch-to-fcm-2

But we need to do some cleanup before we can provide a upstream PR.

**Comment by antoinevg** [comment]
Oh drat, probably a good idea to create an issue on github and co-ordinate before starting work on something like this :-)

I've just pushed `v3.0`:

https://github.com/trigger-corp/trigger.io-parse/commit/15f136ca2d94e1bf97cc8edc5160fa77a2d01992?w=1

In particular, check out the new [`android/build_steps.json`](https://github.com/trigger-corp/trigger.io-parse/blob/master/module/android/build_steps.json). 

I had to make quite a few platform changes to implement  support for the Google Services Gradle plugin and the `google-services.json` file as this now gets compiled into the app binary during build time.

One bonus is that you now have the option to drop `google-services.json` into `<app dir>/assets` instead of `<app dir>/src>` which means that the file won't be shipped with your app bundle. 

There are also some brief migration instructions in the module docs here:

https://trigger.io/modules/parse/v3.0/docs/index.html

`/spend 3 days`
